src/prefect/utilities/urls.py

```diff
@@ -246,7 +249,7 @@ def url_for(
             occurred=obj.occurred.strftime("%Y-%m-%d"), obj_id=obj_id
         )
     else:
-        url = url_format.format(obj_id=obj_id)
+        url = url_format.format(obj_id=obj_id, **additional_format_kwargs)
```

Comment:
I see this raises a `KeyError` if a necessary placeholder isn't filled, but I think a `ValueError` would be more appropriate. Ideally, we'd be able to tell the caller something like, "Unable to generate URL for type {TYPE} due to missing field {FIELD}. Please pass value for {FIELD} into `additional_format_kwargs`."

You can do something like this to extract placeholders that haven't been filled.
```python
>>> from string import Formatter
>>> formatter = Formatter()
>>> [fname for _, fname, _, _ in formatter.parse("work-pools/work-pool/{work_pool_name}/worker/{obj_id}") if fname is not None]
['work_pool_name', 'obj_id']
```